ssz-rs/src/vector.rs

```diff
@@ -118,14 +118,13 @@ where
     T: SimpleSerialize + Default + Clone,
 {
     fn default() -> Self {
+        assert!(N > 0);
+
         let data = vec![T::default(); N];
-        match data.try_into() {
-            Ok(result) => result,
-            // SAFETY: panic
-            // ideally we will not panic here but currently there is no way
-            // to enforce statically that `N` is non-zero with const generics
-            Err((_, err)) => panic!("{err}"),
-        }
+        // SAFETY: panic can't happen because data.len() == N != 0; qed
+        // ideally we will not panic here but currently there is no way
+        // to enforce statically that `N` is non-zero with const generics
+        data.try_into().unwrap_or_else(|(_, err)| panic!("{err}"))
```

Comment:
I find `expect` to be a bit more literate for users, esp when we expect/know this won't panic

```suggestion
        data.try_into().map_err(|(_, err)| err).expect("any Vector can be constructed with default data")
```

1. we need the `map_err` bc I elect to return the "unparsable" data to the caller via the error channel, so we just need to drop the input data (I don't want to require `Debug` on it)
2. then we can just use `expect`